### PR TITLE
feat: accept json in time series endpoints, add fluentbit metrics example

### PR DESCRIPTION
**Pull request by vinaykakade**
## What does this PR do?
- The time series endpoints for searching and appending now accept json,
- Adds an example of using fluentbit to publish CPU metrics to Infino.
## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests. (for non-documentation changes)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -258,19 +252,103 @@ async fn append_log(
...
-  let json_string = serde_json::to_string(&time_series_entry).unwrap();
-  state.queue.publish(&json_string).await.unwrap();
-  let data_point = time_series_entry.data_point;
-  state.tsldb.append_data_point(
-    &time_series_entry.metric_name,
-    &time_series_entry.labels,
-    data_point.get_time(),
-    data_point.get_value(),
-  );
+  Json(ts_json): Json<serde_json::Value>,
```
Comment (savannahar68) [inline src/main.rs:255]: append_ts and append_log function have a lot of similarities. 

Possibility of creating a function to parse json 

```
async fn parse_json(
  State(state): State<Arc<AppState>>,
  Json(json): Json<serde_json::Value>,
  timestamp_key: &str,
) -> Result<Vec<serde_json::Map<String, serde_json::Value>>, (StatusCode, String)> {
  if json.is_object() {
    Ok(vec![json.as_object().unwrap().clone()])
  } else if json.is_array() {
    let objects: Vec<serde_json::Map<String, serde_json::Value>> = json
      .as_array()
      .unwrap()
      .iter()
      .filter_map(|value| value.as_object())
      .cloned()
      .collect();
    Ok(objects)
  } else {
    let msg = format!("Invalid JSON: {}", json);
    error!("{}", msg);
    Err((StatusCode::BAD_REQUEST, msg))
  }
}
```

File: src/main.rs [hunk 2]
```diff
@@ -258,19 +252,103 @@ async fn append_log(
...
+          value_f64 = value.as_u64().expect("Unexpected value type") as f64;
+        } else {
+          error!(
+            "Ignoring value {} for key {} as it is not a number",
+            value, key
+          );
+          continue;
+        }
+
+        state
```
Comment (savannahar68) [inline src/main.rs:344]: I've a doubt - 

If we are pushing to a queue, should we not pull from queue in some background thread and then call append function of TSLDB. [Same for append_log function] 

Or will this be taken up later?

**Review by savannahar68 — APPROVED** [review]
Some refactoring can be taken up. PR looks good to me. Tried fluetbit example on this commit and works fine.

File: src/main.rs (hunk 2 above)
Comment (vinaykakade) [reply src/main.rs:344]: We will be reading from the queue *only* in the disaster scenario at the server startup. Logically, it is our write-ahead-log.

https://github.com/infinohq/infino/issues/27

File: src/main.rs (hunk 1 above)
Comment (vinaykakade) [reply src/main.rs:255]: Thanks for catching, fixed

**Pull request merged**